frontend/src/pages/identityVerification/components/IdentityVerificationForm/IdentityVerificationForm.tsx

```diff
@@ -0,0 +1,250 @@
+  const navigate = useNavigate();
+  const { login } = useAuth();
+
+  const {
+    name,
+    handleNameChange,
+    errorMessage: nameErrorMessage,
+    validated: nameValidated,
+  } = useNameInput();
+
+  const [gender, setGender] = useState<Gender>('남');
+
+  const handleGenderChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    const isGenderType = (
+      value: string,
+    ): value is IdentityVerificationInfo['gender'] => {
+      const genders = ['남', '여'];
+      return genders.includes(value);
+    };
+
+    if (!isGenderType(value)) {
+      return;
+    }
+
+    setGender(value);
+  };
+
+  const {
+    phoneNumber,
+    inputRef,
+    handlePhoneNumberChange: changePhoneNumber,
+  } = useFormattedPhoneNumber();
+
+  const [verificationStep, setVerificationStep] =
+    useState<VerificationStep>('idle');
+
+  const handlePhoneNumberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    changePhoneNumber(e);
+    setVerificationStep('idle');
+  };
+
+  const completeRequest = () => {
+    setVerificationStep('requested');
+  };
+
+  const completeVerification = () => {
+    setVerificationStep('verified');
+  };
+
+  const phoneNumberErrorMessage = getPhoneNumberErrorMessage(phoneNumber);
+
+  const {
+    shouldBlockSubmitByPhoneNumberCheck,
+    handleAuthCodeClick,
+    getFinalPhoneNumberErrorMessage,
+    matchConfirmedPhoneNumber,
+  } = useVerificationCodeRequest({
+    phoneNumber,
+    phoneNumberErrorMessage,
+    completeRequest,
+  });
+
+  const {
+    verificationCode,
+    handleVerificationCodeChange,
+    errorMessage: verificationCodeErrorMessage,
+    validated: verificationCodeValidated,
+  } = useVerificationCodeInput();
+
+  const [submitVerificationErrorMessage, setSubmitVerificationErrorMessage] =
+    useState('');
+
+  const {
+    verificationCodeError,
+    handleAuthCodeVerifyClick,
+    getFinalVerificationCodeErrorMessage,
+    shouldBlockSubmitByVerificationCode,
+  } = useVerificationCodeConfirm({
+    verificationCode,
+    verificationCodeErrorMessage,
+    completeVerification,
+  });
+
+  const getDisplayedVerificationErrorMessage = () => {
+    const errorMessage = getFinalVerificationCodeErrorMessage();
+
+    if (verificationStep !== 'verified') {
+      return errorMessage;
+    }
+
+    return errorMessage === '' ? submitVerificationErrorMessage : errorMessage;
+  };
+
+  const validateForm = () => {
+    const validations = [
+      nameValidated,
+      phoneNumber !== '' && phoneNumberErrorMessage === '',
+      verificationCodeValidated && !verificationCodeError,
+    ];
+
+    return validations.every(Boolean);
+  };
+
+  const verificationRequestButtonEnabled =
+    phoneNumberErrorMessage === '' && phoneNumber !== '';
+
+  const getVerificationButtonEnabled = () => {
+    return (
+      matchConfirmedPhoneNumber &&
+      phoneNumberErrorMessage === '' &&
+      verificationCodeValidated
+    );
+  };
```

Comment:
p5: 단순 boolean 값이라 함수로 안 만들고 계산된 값으로 둬도 될 거 같아 ㅎㅎ 

```suggestion
  const verificationButtonEnabled = 
      matchConfirmedPhoneNumber &&
      phoneNumberErrorMessage === '' &&
      verificationCodeValidated
```

Reply:
[refactor: boolean 리턴 값을 가지는 함수 변수로 변경](https://github.com/woowacourse-teams/2025-Fit-toring/pull/827/commits/39a9a0ea0420f696747dc36fdd73a32b3bdaa815)
👍👍👍